blog post about grpc stacks

carl-mastrangelo opened the pull request:
cc: @menghanl @ncteisen

srini100 commented on _posts/2018-11-30-grpc-stacks.md:
```diff
@@ -0,0 +1,40 @@
+<!--more-->
+
+
+There are three main stacks in gRPC: C-core, Go, and Java.  Most of the languages are thin wrappers on top of the [C-based](https://github.com/grpc/grpc/tree/v1.16.1/src/core) gRPC core library:
```
Link to master branch as 1.16.1 will be old soon.

carl-mastrangelo replied:
I had picked a tag because links tend to break over time (re-orging code).  Done, though.

srini100 commented on _posts/2018-11-30-grpc-stacks.md:
```diff
@@ -0,0 +1,40 @@
+
+<p><img src="https://grpc.io/img/grpc-core-stack.svg" alt="gRPC Core Stack" style="max-width: 800px" /></p>
+
+For example, a Python application calls into the generated Python stubs.  These call pass through interceptors, and into the wrapping library where the call is translated into C calls.  The gRPC C-core will encode the RPC as HTTP/2, optionally encrypt the data with TLS, and then pass write it to the network.
```
These *calls* ...
...where the *calls are* ...

srini100 replied:
then ~pass~ write

carl-mastrangelo replied:
Done.

srini100 commented on _posts/2018-11-30-grpc-stacks.md:
```diff
@@ -0,0 +1,40 @@
+
+<p><img src="https://grpc.io/img/grpc-java-stack.svg" alt="gRPC Java Stack" style="max-width: 800px" /></p>
+
+Again, the structure is a little different.  Java supports HTTP/2, QUIC, and In Process like the C-core.  Unlike the C-Core though, applications commonly can bypass the generated stubs and interceptors, and speak directly to the Java Core library.  Each structure is slightly different based on the needs of each language implementation of gRPC.
```
Is QUIC supported directly or is it via Cronet? If Cronet then the diagram needs to change a include TCP under Cronet.

carl-mastrangelo replied:
From talking to @ncteisen I thought there was built in support not from Cronet.

hsaliak replied:
@vjpai as far as I am aware we do not have a non cronet QUIC transport..  IIUC We do not have server side support of quic either, but rely on a translating proxy in between...

carl-mastrangelo commented:
@srini100 PTAL

carl-mastrangelo commented on _posts/2018-11-30-grpc-stacks.md:
```diff
@@ -0,0 +1,40 @@
+
+<p><img src="https://grpc.io/img/grpc-core-stack.svg" alt="gRPC Core Stack" style="max-width: 800px" /></p>
+
+For example, a Python application calls into the generated Python stubs.  These calls pass through interceptors, and into the wrapping library where the call is translated into C calls.  The gRPC C-core will encode the RPC as HTTP/2, optionally encrypt the data with TLS, and then write it to the network.
```
Done.

hsaliak commented on _posts/2018-11-30-grpc-stacks.md:
```diff
@@ -0,0 +1,40 @@
+<!--more-->
+
+
+There are three main stacks in gRPC: C-core, Go, and Java.  Most of the languages are thin wrappers on top of the [C-based](https://github.com/grpc/grpc/tree/master/src/core) gRPC core library:
```
C++ based? but a restricted subset of C++? How do we want to word this? @vjpai any suggestions here?

sanjaypujare commented on _posts/2018-12-11-grpc-stacks.md:
```diff
@@ -0,0 +1,43 @@
+company-link: https://www.google.com
+---
+
+Here is a high level overview of the gRPC Stacks.  Each of the **10** default languages supported by gRPC has multiple layers, allowing you to customize what pieces you want in your application.
```
gRPC Stacks => gRPC Language Stacks

sanjaypujare commented:
Looks good. In the next version it will be good to include a feature comparison matrix or table for the various language stacks so one can quickly see the Cronet and QUICK support is offered by core and Java but not by Go.

carl-mastrangelo commented:
@sanjaypujare Agree on the matrix, though these are somewhat time consuming to make.  (and keep up to date).  

I sent a PR addressing your comments in #793
